src/main/java/util/RefreshTimer.java

```diff
@@ -0,0 +1,122 @@
+    public static final int DEFAULT_REFRESH_PERIOD_IN_MINS = 1;
+
+    public RefreshTimer(String name, int period, Consumer<Integer> onTick, Runnable onTimeout,
+                        TimeUnit timeUnit) {
+        super(name, period, onTick, onTimeout, timeUnit);
+    }
+
+    /**
+     * Concurrency may cause pause method to trigger multiple times. Checking is done for fail-safe purpose.
+     */
+    @Override
+    public synchronized void pause() {
+        if (!isPaused()){
+            super.pause();
+        }
+    }
+
+    /**
+     * Concurrency may cause resume method to trigger multiple times. Checking is done for fail-safe purpose.
+     */
+    @Override
+    public synchronized void resume() {
+        if (isPaused()) {
+            super.resume();
+        }
+    }
+
+    /**
+     * Computes the TickerTimer period that is used for refreshing the issues periodically
+     * Assumes future refreshes will take the same number of API calls as the last refresh and find out the refresh
+     * duration that will spread out the refreshes until the next api quota top up.
+     *
+     * For some cases where the refresh rate is equal to the remainingTimeInMins,
+     * BUFFER_TIME is added to ensure that the next refresh happens after the apiQuota renewal.
+     *
+     * PMD is suppressed to enable necessary parenthesis.
+     *
+     * @param apiQuota The remaining allowed api request until the next api quota renewal.
+     *                 Pre-condition: >= 0
+     * @param remainingTimeInMins The remaining time left until the next api quota renewal.
+     *                           Pre-condition: >= 0
+     * @param apiCallsUsedInPreviousRefresh The amount of api used in the last api refresh.
+     *                                      Pre-condition: >= 0
+     * @param apiQuotaBuffer The amount of api calls that is set aside for manual operations by the user.
+     *                       Pre-condition: >= 0
+     * @param minRefreshPeriod The minimal refresh period that will be used.
+     *                         There are 3 conditions that this will be used.
+     *                         1) During application initialisation, when apiCallsUsedInPreviousRefresh is zero.
+     *                         2) When computed refresh rate is < minRefreshPeriod.
+     *                         3) When remainingTimeInMins = 0. This indicates that refreshes can occur in the
+     *                            nearest time possible.
+     *                            Recommended value : RefreshTimer.DEFAULT_REFRESH_PERIOD_IN_MINS
+     *                         Pre-condition: > 0
+     * @return Returns computed refresh period.
+     */
+    @SuppressWarnings("PMD")
+    public static long computeRefreshTimerPeriod(int apiQuota, long remainingTimeInMins,
+                                                 int apiCallsUsedInPreviousRefresh,
+                                                 int apiQuotaBuffer, int minRefreshPeriod) {
+
```

Comment:
It's a method with clear inputs and only one output. All inputs are numbers. You can use test case design heuristics (e.g. equivalence classes, boundary values, etc.) covered in CS2103 when testing this.